### PR TITLE
refactor: remove use of `try-with`

### PR DESCRIPTION
**Pull request by jaschdoc**
Depends on https://github.com/flix/flix/pull/9409
Related to https://github.com/flix/flix/issues/9214

### CODE REVIEW

File: main/test/flix/Test.Handler.NewObject.flix [hunk 1]
```diff
@@ -30,9 +30,9 @@ mod Test.Handler.NewObject {
 
     def newGenAskRunnable(r: Ref[Int32, rc]): Runnable \ IO = new Runnable {
         def $run(_this: Runnable): Unit \ { rc } =
-            try {
+            run {
                 Ask.ask(
-                    try {
+                    run {
```
Comment (jaschdoc) [inline main/test/flix/Test.Handler.NewObject.flix:35]: Weird, parsing does not allow nested run-with

Comment (magnus-madsen) [reply main/test/flix/Test.Handler.NewObject.flix:35]: Uh oh. You may have to check my stupid code!

Comment (jaschdoc) [reply main/test/flix/Test.Handler.NewObject.flix:35]: I will revert these to errors to `try-with` for now and fix it when we also remove the old `try-with` syntax completely (it's on 2 places in the entire code base that needs to be updated)

Comment (magnus-madsen) [reply main/test/flix/Test.Handler.NewObject.flix]: Can we not fix it now? Separately?

Comment (magnus-madsen) [reply main/test/flix/Test.Handler.NewObject.flix]: Its seems like a pretty bad bug, so would be good to have a fix in.

Comment (jaschdoc) [reply main/test/flix/Test.Handler.NewObject.flix]: True I have been working on it for 45 minutes now at it is not straightforward why this happens

Comment (magnus-madsen) [reply main/test/flix/Test.Handler.NewObject.flix]: Gosh.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**